_posts/en/newsletters/2020-01-29-newsletter.md

```diff
@@ -0,0 +1,215 @@
+  proposal.  If the proposal is adopted, users will be able to create
+  UTXOs that can only be spent by a specific transaction or set of
+  transactions, providing a type of [covenant][topic covenants].  This is
+  useful in protocols that temporarily keep transactions offchain but
+  which need to provide strong guarantees that, when the transaction
+  goes onchain, it will be exactly the transaction the receiver
+  expected.
+
+- [BIPs #876][] assigns BIPs to the elements of the schnorr-taproot
+  proposal:
+
+    - [BIP340][] is assigned to "Schnorr Signatures for secp256k1,"
+      which describes a signature scheme compatible with the secp256k1
+      [elliptic curve][] used by Bitcoin.  The signatures are
+      compatible with batch verification and with key and signature
+      aggregation schemes such as [MuSig][topic musig].  Schnorr
+      signatures are made available for use in the following two BIPs.
+      For more information, see the BIP or [schnorr signatures][topic
+      schnorr signatures].
+
+    - [BIP341][] is assigned to "Taproot: SegWit version 1 spending
+      rules," which describes part of a soft fork proposal for allowing
+      users to pay schnorr-style public keys that can be spent either
+      using a schnorr-style signature or proof that the key committed to
+      a particular script via a merkle tree (along with proof that the
+      script's conditions were fulfilled).  For details, see the BIP or
+      [taproot][topic taproot].
+
+    - [BIP142][] is assigned to "Validation of Taproot Scripts", which
+      describes the rules for evaluating a script used in combination
+      with taproot (a *tapscript*).  Almost all operations in tapscript
+      are the same as legacy Bitcoin Script, but a few are different.
+      For existing users upgrading to tapscript, the most significant
+      change is that all signature-checking opcodes (e.g. `OP_CHECKSIG`)
+      use schnorr public keys and signatures; also noteworthy is that
+      `OP_CHECKMULTISIG` has been removed; script authors can instead
+      use a new `OP_CHECKSIGADD` opcode or otherwise redesign their
+      scripts.  A few other new rules affect users or rarely-used
+      features.  Additionally, tapscript includes several new features
+      designed to make future soft fork upgrades of its scripting
+      language easier.  For details, see the BIP or [tapscript][topic
+      tapscript].
+
+    {% comment %}<!--
+    ~/repos/bips$ git shortlog -s 6a80232...97ef678 | wc -l
+    31
+    # One name is duplicated
+
+    $ git log --oneline 6a80232...97ef678 | wc -l
+    426
+    -->{% endcomment %}
+
+    Although many merges to the BIPs repository include contributions
+    from several different people, this merge had more contributors than
+    any we've seen before: it included content and edits from 30
+    different people in 426 commits and its credits thank several other
```

Comment:
I only count 163 commits (from merge base commit 24eddbb) and 28 authors (2 duplicates):

```
→ git log 24eddbb.. | grep Author | cut -f2 -d ":" |sort |uniq
 Adam Gibson <ekaggata@gmail.com>
 Agis Anastasopoulos <a@xz0.org>
 andrewtoth <andrewstoth@gmail.com>
 Anthony <pyskell@users.noreply.github.com>
 Anthony Towns <aj@erisian.com.au>
 Bryan Bishop <kanzure@gmail.com>
 codeShark149 <rajarshi149@gmail.com>
 Dev Random <devrandom@users.noreply.github.com>
 Devrandom <miron@basezero.com>
 Dmitry Petukhov <dp@simplexum.com>
 Elichai Turkel <elichai.turkel@gmail.com>
 Fabian Jahr <fabjahr@gmail.com>
 Gregory Sanders <gsanders87@gmail.com>
 Hennadii Stepanov <32963518+hebasto@users.noreply.github.com>
 JamesC <james.chiangwu@gmail.com>
 John Newbery <john@johnnewbery.com>
 Jonas Nick <jonasd.nick@gmail.com>
 Jon Atack <jon@atack.com>
 Kalle Rosenbaum <kalle@rosenbaum.se>
 LaurentMT <LaurentMT@users.noreply.github.com>
 Mark B Lundeberg <36528214+markblundeberg@users.noreply.github.com>
 Matthew Zipkin <pinheadmz@gmail.com>
 Max Hillebrand <30683012+MaxHillebrand@users.noreply.github.com>
 Orfeas Litos <o.thyfronitis@ed.ac.uk>
 Orfeas Stefanos Thyfronitis Litos <o.thyfronitis@ed.ac.uk>
 Pieter Wuille <pieter.wuille@gmail.com>
 stefanwouldgo <stefan@sblbs.de>
 Thomas Kerin <me@thomaskerin.io>
 Tim Ruffing <crypto@timruffing.de>
 Tim Ruffing <tim@timruffing.de>
→ git log 24eddbb.. | grep Author | cut -f2 -d ":" |sort |uniq | wc -l
30
```

Reply:
Confirmed.  Looks like the major difference was my script was counting merges on sipa's branch.

```text
harding@ganymede:~/repos/bips$ git log --oneline --no-merges  802520e...9cf4038 | wc -l
163
harding@ganymede:~/repos/bips$ git shortlog -s  802520e...9cf4038 | wc -l
30
```

Also, I missed the second duplicate author (I see them now).

Thanks for verifying!